cas-server-webapp-support/src/main/java/org/jasig/cas/util/CasLoggerContextInitializer.java

```diff
@@ -0,0 +1,164 @@
+
+    private ServletContextListener loggerContext;
+
+    private final String loggerContextPackageName;
+
+    private final Resource logConfigurationFile;
+
+    private final String logConfigurationField;
+
+    /**
+     * Instantiates a new Cas logger context initializer.
+     */
+    protected CasLoggerContextInitializer() {
+        this.loggerContext = null;
+        this.loggerContextPackageName = null;
+        this.logConfigurationField = null;
+        this.logConfigurationFile = null;
+    }
+
+    /**
+     * Instantiates a new Cas logger context initializer.
+     *
+     * @param loggerContextPackageName the logger context package name
+     * @param logConfigurationFile the log configuration file
+     * @param logConfigurationField the log configuration field
+     */
+    public CasLoggerContextInitializer(@NotNull final String loggerContextPackageName,
+                                       @NotNull final Resource logConfigurationFile,
+                                       @NotNull final String logConfigurationField) {
+        this.loggerContextPackageName = loggerContextPackageName;
+        this.logConfigurationField = logConfigurationField;
+        this.logConfigurationFile = logConfigurationFile;
+    }
+
+    /**
+     * Initialize the logger by decorating the context
+     * with settings for the log file and context.
+     * Calls the initializer of the logging framework
+     * to start the logger.
+     */
+    private void initialize() {
+        try {
+            if (!INITIALIZED.get() && this.loggerContext != null) {
+                final ServletContextEvent event = new ServletContextEvent(this.context);
+                this.loggerContext.contextInitialized(event);
+                LOGGER.debug("Initialized logging context via [{}]. Logs will be written to [{}]",
+                        this.loggerContext.getClass().getSimpleName(),
+                        this.logConfigurationFile);
+                INITIALIZED.set(true);
+            }
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Destroys all logging hooks and shuts down
+     * the logger.
+     */
+    @PreDestroy
+    public void destroy() {
+        try {
+            if (INITIALIZED.get() && this.loggerContext != null) {
+                final ServletContextEvent event = new ServletContextEvent(this.context);
+                LOGGER.debug("Destroying logging context and shutting it down");
+                this.loggerContext.contextDestroyed(event);
+            }
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Prepares the logger context. Locates the context and
+     * sets the configuration file.
+     * @return the logger context
+     */
+    private ServletContextListener prepareAndgetContextListener() {
+        try {
+            if (StringUtils.isNotBlank(this.loggerContextPackageName)) {
```

Comment:
I still think this is too complicated if we "just" initialize the ```Log4ContextListener``` with the appropriate file configuration path: why do we have a ```loggerContextPackageName``` property? Isn't *org.apache.logging.log4j.core.web* the right package?

Isn't it worrying to have Spring not support log4j v2? As a mesure of log4j v2 adoption and support...

What Google trends says: http://www.google.com/trends/explore#q=log4j%202%2C%20logback&cmpt=q&tz=